qcodes/instrument/parameter.py

```diff
@@ -2766,3 +2766,29 @@ def expand_setpoints_helper(parameter: ParameterWithSetpoints,
         data = results
     res.append((parameter, data))
     return res
+
+
+class AbstractParameter(Parameter):
+    def __init__(self, name: str,
+                 instrument: Optional['InstrumentBase'] = None,
+                 **kwargs: Any):
+        """
+        An abstract parameter is meant to define an abstract interface,
+        much like abc.abstractmethod. This is useful for defining instrument
```

Comment:
question - unlike the ABC infrastructure, the current implementation of the AbstractParameter is such that it is not enforced to be "implemented" at the class instantiation time. the only thing that there is is the set/get which raise exception. questions are:
- is this good enough for defining "abstract instrument interfaces"?
- if not, can we come up with something better (that perhaps does not use metaclasses ;) )?

Reply:
I think this is hard to do in a clean way since there are so many use-cases of parameter.

Normally one would make `get_raw` and `set_raw` abstract methods and then 
enforce that it is illegal to instantiate a class where these are not overwritten but we have to support the following usecases. 

* get only parameter. `set_raw` is not implemented
* set only parameter. `get_rwa` is not implemented
* set/get cmd passed as a string to `__init__`